Functional.Primitives.Extensions/ResultExtensions.cs

```diff
@@ -113,6 +225,42 @@ public static class ResultExtensions
 		public static Task Apply<TSuccess, TFailure>(this Task<Result<TSuccess, TFailure>> result, Action<TSuccess> success)
 			=> result.Do(success, _ => { });
 
+		public static void Apply<TSuccess, TFailure>(this Result<Option<TSuccess>, TFailure> result, Action<TSuccess> successSome, Action successNone, Action<TFailure> failure)
```

Comment:
I feel there are enough extensions here that could create some ambiguity that it might be worth thinking of a general and consistent pattern for the naming. OptionSelect, OptionBind, OptionApply? I don't know... any thoughts?

Reply:
The code I originally ported these from used the names `SelectUsingOptionValue`, `SelectUsingOptionValueOntoOptionValue`, `BindUsingOptionValue`...  but I wasn't a huge fan of those.  `Do` and `Apply` should be renamed to avoid collisions, for sure.